src/component.js

```diff
@@ -0,0 +1,289 @@
+    module.exports = factory(require('jquery'), window);
+  } else {
+    // Browser global
+    window.Component = factory(window.$, window);
+  }
+
+}(function($, window) {
+
+  var config = {
+    debug: false
+  }
+
+  /*
+   * Add remove event to jQuery.
+   *
+   * Copy from jquery.ui
+   * Need include component after jquery ui
+   */
+
+  if ($ && !$.ui) {
+    $.cleanData = (function( orig ) {
+      return function( elems ) {
+        var events, elem, i;
+        for ( i = 0; ( elem = elems[ i ] ) != null; i++ ) {
+         // Only trigger remove when necessary to save time
+          events = $._data( elem, "events" );
+          if ( events && events.remove ) {
+            $( elem ).triggerHandler( "remove" );
+          }
+        }
+        orig( elems );
+      };
+    })($.cleanData);
+  }
+
+  var extend = function(Parent, proto) {
+    // props in _superProps
+
+    var F = function() {
+      return Parent.apply(this, arguments);
+    }
+
+    var _super = $.extend(true, {}, Parent.prototype);
+
+    F.prototype = Object.create(_super);
+    F.prototype.constructor = Parent;
+
+    var props = {};
+
+    for (var key in proto) {
+      var value = proto[key];
+      if (!proto.hasOwnProperty(key)) continue;
+
+      if (typeof value === 'function') {
+        F.prototype[key] = _super[key] ? (function(key, value) {
+          return function() {
+            // Warning :)
+            this._super = _super[key];
+            var result = value.apply(this, arguments);
+            this._super = undefined;
+            return result;
+          };
+        })(key, value) : value;
+      } else {
+        // need support for array and object
+        props[key] = value;
+      }
+    }
+
+    if (_super._superProps) {
+      _super._superProps = $.extend(true, {}, _super._superProps, props);
+    } else {
+      _super._superProps = props;
+    }
+
+    return F;
+  };
+
+  var Registry = {
+    components: {},
+    mixins: {}
+  }
+
+  var filters = [];
+
+  function Core($block, options) {
+    $.extend(true, this, this._superProps, options);
+
+    this.$block = $block;
+
+    if (typeof this.prevent == 'function' && this.prevent()) {
+      if (config.debug) console.warn('Component %s is prevented.', this._namespace);
+      return;
+    }
+
+    for (var i = 0; i < filters.length; i++) {
+      filters[i].call(this);
+    }
+
+    this.init();
+
+    return this;
+  }
+
+  Core.prototype = {
+    init: function() {},
+
+    _elName: function(name) {
+      return ['.js-', this._namespace, name[0].toUpperCase() + name.slice(1)].join('');
+    },
+
+    _componentName: function(name) {
+      return '[data-component~="' + name + '"]';
+    },
+
+    trigger: function(event, data) {
+      return this.$block.trigger(event, data);
+    },
+
+    dispatch: function(event, data) {
+      return this.$block.parents('[data-component]').triggerHandler(event, data);
+    },
+
+    broadcast: function(event, data) {
```

Comment:
return this.$block.find('[data-component]').each(function() {
  $(this).triggerHandler(event, data);
});